src/sankoff_handler.hpp

```diff
@@ -12,116 +12,159 @@
 #include "node.hpp"
 #include "driver.hpp"
 
+// All partial vectors for all nodes at one site
+using SankoffPartial = Eigen::Matrix<double, Eigen::Dynamic, 4>;
+// Each SankoffPartial represents a calculations for one site
+using SankoffPartialVec = std::vector<SankoffPartial>;
+
 class SankoffHandler {
-    public:
-        // DNA assumption
-        static constexpr size_t state_count = 4;
-        // bifurating tree assumption
-        static const size_t child_count = 2;
-
-    SankoffHandler(SitePattern site_pattern) {
-        Assert(site_pattern.SequenceCount() == site_pattern.TaxonCount(), "Error in SankoffHandler constructor 1: Every sequence should be associated with a node.");
-        mutation_costs_ = SankoffMatrix();
-        site_pattern_ = std::move(site_pattern);
-    };
-    SankoffHandler(Eigen::Matrix<double, state_count, state_count> cost_matrix, SitePattern site_pattern) {
-        Assert(site_pattern.SequenceCount() == site_pattern.TaxonCount(), "Error in SankoffHandler constructor 2: Every sequence should be associated with a node.");
-        mutation_costs_ = SankoffMatrix(cost_matrix);
-        site_pattern_ = std::move(site_pattern);
-    };
-
-    SankoffHandler(SankoffMatrix sankoff_matrix, SitePattern site_pattern) {
-        Assert(site_pattern.SequenceCount() == site_pattern.TaxonCount(), "Error in SankoffHandler constructor 3: Every sequence should be associated with a node.");
-        mutation_costs_ = std::move(sankoff_matrix);
-        site_pattern_ = std::move(site_pattern);
-    };
-    
-    EigenVectorXd TotalPPartial(Eigen::Matrix<double, Eigen::Dynamic, state_count> left, Eigen::Matrix<double, Eigen::Dynamic, state_count> right, size_t node_id);
-    std::vector<Eigen::Matrix<double, Eigen::Dynamic, state_count>> BlankPartials(size_t node_count);
-
-    Eigen::Matrix<double, Eigen::Dynamic, state_count> GenerateLeavesPartials(const size_t node_count, const size_t pattern_idx);
-
-    EigenVectorXd ParentPartial(EigenVectorXd child_partials);
-
-    // Calculates Sankoff partials for all nodes at all sites.
-    std::tuple<std::vector<Eigen::Matrix<double, Eigen::Dynamic, SankoffHandler::state_count>>, std::vector<Eigen::Matrix<double, Eigen::Dynamic, SankoffHandler::state_count>>, std::vector<Eigen::Matrix<double, Eigen::Dynamic, SankoffHandler::state_count>>> RunSankoff(Node::NodePtr topology);
-
-    // Calculates parsimony score on given node across all sites.
-    double ParsimonyScore(std::tuple<std::vector<Eigen::Matrix<double, Eigen::Dynamic, SankoffHandler::state_count>>, std::vector<Eigen::Matrix<double, Eigen::Dynamic, SankoffHandler::state_count>>, std::vector<Eigen::Matrix<double, Eigen::Dynamic, SankoffHandler::state_count>>> all_partials, size_t node_id);
-
-  private:
-    SankoffMatrix mutation_costs_;
-    SitePattern site_pattern_;
+ public:
+  static inline const size_t partial_type_count = 3;
```

Comment:
All of these members should end with an underscore (sorry to miss this the first time around).